data/virt_autotest/virt_logs_collector.sh

```diff
@@ -0,0 +1,427 @@
+         -nocase "mistake|wrong|fault|error|fail|exception|not*found|timed*out" {puts "Can not establish virsh console to \${guest_domain}\r"; set ret_result 1}
+      }
+      expect {
+         -nocase "login: $"  {send "root\r"; set ret_result 0; exp_continue -continue_timer}
+         -nocase "password:" {send "novell\rcd ~\r"; exp_continue -continue_timer}
+         -nocase "mistake|wrong|fault|error|fail|exception|not*found|timed*out" {puts "Can not login virsh console to \${guest_domain}\r"; set ret_result 1}
+      }
+
+      if { \${ret_result} == 0 } {
+         set timeout 600
+         expect "~ #"
+         send "mkdir -p \${logs_folder};export time_stamp=\`date \'+%Y%m%d%H%M%S\'\`\r"
+         expect "~ #"
+         if { \${extra_logs} == {support_config} } {
+            send "rm -f -r \${logs_folder}/*supportconfig*\r"
+            send "supportconfig -t \${logs_folder} -B guest_\${guest_transformed}_supportconfig_\\\${time_stamp}\r"
+         }
+         if { \${extra_logs} != {support_config} && \${extra_logs} != "" } {
+            send "mkdir -p \${logs_folder}/guest_\${guest_transformed}_extra_logs;cp -r -f \${extra_logs} \${logs_folder}/guest_\${guest_transformed}_extra_logs\r"
+         }
+         expect {
+            "~ #" {send "export ret=\\\$?;cd /tmp;(if \[\[ \\\$ret -ne 0 ]];then printenv USER;fi)\r"; exp_continue -continue_timer}
+            "root" {puts "Can not collect \${extra_logs} from \${guest_domain}\r"; set ret_result 1;  exp_continue -continue_timer}
+            "tmp #" {send "cd ~\r"; send "exit\r"}
+         }
+      }
+
+      if { \${ret_result} == 0 && \${extra_logs} != "" } {
+         expect -nocase "login: $" {puts "Successfully collected \${extra_logs} from \${guest_domain}\r"; exit \${ret_result}}
+      }
+
+      if { \${ret_result} == 1 && \${extra_logs} != "" } {
+         expect -nocase "login: $" {puts "Will try to collect \${extra_logs} from \${guest_domain} again\r"}
+      }
+
+      if { \${extra_logs} == "" } {
+         expect -nocase "login: $" {puts "Nothing to collect from \${guest_domain}. Please specify something.\r"}
+      }
+
+      set retry_times [expr \${retry_times}-1]
+      close -i \${spawn_id}
+}
+exit \${ret_result}
+EOF
+
+        echo -e "${expfile} to be executed is as below:"
+        cat ${expfile}
+  
+	for procid in `ps aux | grep  ${expfile} | grep -v "grep" | awk '{print $2}'`;do
+            kill -9 ${procid}
+	done
+	for procid in `ps aux | grep "virsh console" | grep -v "grep" | awk '{print $2}'`;do
+	    kill -9 ${procid}
+	done
+	echo -e "expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}""
+	expect ${expfile} "${hypervisor}" "${guest_domain}" "${guest_transformed}" "${logs_folder}" "${extra_logs}"
+        if [[ $? == 0 ]];then
+           echo -e "${expfile} returned with success. Successfully collected ${extra_logs} via ${guest_domain} console."
+           rm -f ${expfile}
+	   return 0
```

Comment:
This line is not aligned. 

BTW, seems the indent is not unified in this file, some are 2 blanks, some are 3, and some are 6.